libs/platform/src/lib/components/base.ts

```diff
@@ -58,7 +69,36 @@ export abstract class BaseComponent {
         this._disabled = disabled;
     }
 
-    constructor(protected _cd: ChangeDetectorRef) {}
+    constructor(protected _cd: ChangeDetectorRef) {
+        const injector = PlatformConfig.getInjector();
+        this._contentDensityService = injector.get(ContentDensityService);
+        this._router = injector.get(Router);
+        this._router.events.subscribe(event => {
+            if (event instanceof NavigationEnd) {
+                this._setupDensitySubscriptions();
+            }
+        })
+    }
+
+    /** @hidden */
+    ngOnInit(): void {
+        this._setupDensitySubscriptions();
+    }
+
+    /** @hidden */
+    ngOnDestroy(): void {
+        this._subscriptions.unsubscribe();
+    }
+
+    /** @hidden */
+    private _setupDensitySubscriptions(): void {
+        if (this._contentDensity === null && this._contentDensityService) {
+            this._subscriptions.add(this._contentDensityService.contentDensity.subscribe(density => {
+                this.contentDensity = density;
+                this.detectChanges();
```

Comment:
can this be changed to `markForCheck`? Dynamic page tabbed [example](https://deploy-preview-4782--fundamental-ngx.netlify.app/#/core/dynamic-page#tabbed) adds additional subscriptions and they don't take affect with `this.detectChanges()` here triggering change detection immediately thereby breaking the example.